js/compiler/compiler.go

```diff
@@ -124,7 +132,28 @@ func (c *Compiler) Transform(src, filename string, inputSrcMap []byte) (code str
 		return
 	}
 
-	code, srcMap, err = c.babel.transformImpl(c.logger, src, filename, c.Options.SourceMapLoader != nil, inputSrcMap)
+	sourceMapEnabled := c.Options.SourceMapLoader != nil
+	maxSrcLenForBabelSourceMapOnce.Do(func() {
+		// TODO: drop this code and everything it's connected to when babel is dropped
+		v := os.Getenv(maxSrcLenForBabelSourceMapVarName)
+		if len(v) > 0 {
+			i, err := strconv.Atoi(v)
+			if err != nil {
+				c.logger.Warnf("Tried to parse %q from %s as integer but couldn't %s\n",
+					v, maxSrcLenForBabelSourceMapVarName, err)
+				return
+			}
+			maxSrcLenForBabelSourceMap = i
+		}
+	})
+	if sourceMapEnabled && len(src) > maxSrcLenForBabelSourceMap {
+		sourceMapEnabled = false
+		c.logger.Warnf("the source for `%s` needs to go through babel but is over %d bytes. "+
+			"For performance reasons sourcemaps support will be disabled for this particular file.",
```

Comment:
```suggestion
			"For performance reasons source map support will be disabled for this particular file.",
```
for consistency with other logs